### PR TITLE
Sample config file changes

### PR DESCRIPTION
**Pull request by aaronlehmann**
Rename config.yml to dev-config.yml

Add example-config.yml, a simple configuration file for the official
This was originally made for the the distribution-library-image repo,
but is being moved here to make sure it stays in sync.

Update Dockerfile and docs for the rename.

Signed-off-by: Aaron Lehmann <aaron.lehmann@docker.com>

### CODE REVIEW

**Comment by dmp42** [comment]
LGTM

**Comment by RichardScothern** [comment]
LGTM

File: Dockerfile [hunk 1]
```diff
@@ -15,4 +15,4 @@ RUN make PREFIX=/go clean binaries
 VOLUME ["/var/lib/registry"]
 EXPOSE 5000
 ENTRYPOINT ["registry"]
-CMD ["cmd/registry/config.yml"]
```
Comment (stevvooe) [inline Dockerfile]: This was a documented mount point. Can't we just move this into place with image build?

File: docs/configuration.md [hunk 2]
```diff
@@ -1552,7 +1552,7 @@ The above configures the registry instance to run on port `5000`, binding to
 verbose.
 
 A similar simple configuration is available at
-[config.yml](https://github.com/docker/distribution/blob/master/cmd/registry/config.yml).
+[example-config.yml](https://github.com/docker/distribution/blob/master/cmd/registry/example-config.yml).
```
Comment (stevvooe) [inline docs/configuration.md]: `config-example.yml`.

File: Dockerfile (hunk 1 above)
Comment (aaronlehmann) [inline Dockerfile]: Yes, I'll submit a PR to copy dev-config.yml to config.yml at image build to preserve the old location.

File: docs/configuration.md (hunk 2 above)
Comment (aaronlehmann) [inline docs/configuration.md]: The file is named example-config.yml; are you saying it should be renamed?

Comment (stevvooe) [inline docs/configuration.md]: Was that already there? It should be renamed to start with config. That way, config files sort together.

File: Dockerfile (hunk 1 above)
Comment (aaronlehmann) [inline Dockerfile]: https://github.com/docker/distribution/pull/793

File: docs/configuration.md (hunk 2 above)
Comment (aaronlehmann) [inline docs/configuration.md]: OK, once #793 is merged, I'll open a PR renaming the files to start with `config-`.

Comment (aaronlehmann) [inline docs/configuration.md]: https://github.com/docker/distribution/pull/795

**Pull request merged**